### PR TITLE
`Communication`: Update thread when switching reply context

### PR DESCRIPTION
**Pull request by badkeyy**
#### General

- [ ] I tested **all** changes and their related features with **all** corresponding user types on a test server.
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/development-process.html#naming-conventions-for-github-pull-requests).

#### Client
- [x] **Important**: I implemented the changes with a very good performance, prevented too many (unnecessary) REST calls and made sure the UI is responsive, even with large data (e.g. using paging).
- [x] I **strictly** followed the principle of **data economy** for all client-server REST calls.
- [x] I **strictly** followed the [client coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] Following the [theming guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-design/), I specified colors only in the theming variable files and checked that the changes look consistent in both the light and the dark theme.
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I added `authorities` to all new routes and checked the course groups for displaying navigation elements (links, buttons).
- [x] I documented the TypeScript code using JSDoc style.
- [x] I added multiple screenshots/screencasts of my UI changes.
- [x] I translated all newly inserted strings into English and German.

### Motivation and Context
When using the “Show X Replies” button in the Communication section to open threads, switching between threads does not update the displayed chat text in the thread view. Instead, the chat text from the first opened thread remains visible, which can lead to confusion.
Fixes https://github.com/ls1intum/Artemis/issues/9754 and https://github.com/ls1intum/Artemis/issues/9803

### Description
After many hours of debugging I found that a ngOnChanges lifecycle hook in PostingContentPartComponent was missing. I added the hook and it now the threads work as expected.

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Course with existing communications

1. Log in to Artemis
2. Navigate to Course Communication section
3. Click on “Show X Replies” to open a thread.
4. Switch to another thread using the same method.
5. Observe that the chat text in the thread view does update 

[](https://artemis-test1.artemis.cit.tum.de)
[](https://artemis-test2.artemis.cit.tum.de)
[](https://artemis-test3.artemis.cit.tum.de)
[](https://artemis-test4.artemis.cit.tum.de)
[](https://artemis-test5.artemis.cit.tum.de)
[](https://artemis-test6.artemis.cit.tum.de)
[](https://artemis-test9.artemis.cit.tum.de)

#### Performance Review

- [ ] I (as a reviewer) confirm that the client changes (in particular related to REST calls and UI responsiveness) are implemented with a very good performance even for very large courses with more than 2000 students.
- [ ] I (as a reviewer) confirm that the server changes (in particular related to database calls) are implemented with a very good performance even for very large courses with more than 2000 students.
#### Code Review
- [ ] Code Review 1
- [ ] Code Review 2
#### Manual Tests
- [ ] Test 1
- [ ] Test 2
#### Exam Mode Test
- [ ] Test 1
- [ ] Test 2
#### Performance Tests
- [ ] Test 1
- [ ] Test 2

### Screenshots

video

## Summary by CodeRabbit

- **New Features**
	- Enhanced the `PostingContentPart` component to react dynamically to changes in input properties, ensuring content is processed and updated in real-time.

### CODE REVIEW

File: src/main/webapp/app/shared/metis/posting-content/posting-content-part/posting-content-part.components.ts [hunk 1]
```diff
@@ -54,10 +54,16 @@ export class PostingContentPartComponent implements OnInit {
         private accountService: AccountService,
     ) {}
 
-    ngOnInit() {
+    ngOnInit(): void {
         this.processContent();
     }
 
+    ngOnChanges(changes: SimpleChanges): void {
+        if (changes['postingContentPart'] && !changes['postingContentPart'].firstChange) {
```
Comment (BBesrour) [inline src/main/webapp/app/shared/metis/posting-content/posting-content-part/posting-content-part.components.ts:62]:
```suggestion
        if (changes.postingContentPart && !changes.postingContentPart.firstChange) {
```
We mostly use this style in client code

**Review by BBesrour — APPROVED** [review]
code lgtm. Just a small nitpick

**Comment by SimonKaran13** [comment]
For some reason, adding the deployment labels does not trigger a deployment

**Comment by badkeyy** [comment]
> For some reason, adding the deployment labels does not trigger a deployment


Yea this seems to be a general problem right now...

**Review by BBesrour — APPROVED** [review]

**Comment by BBesrour** [comment]
code

**Comment by badkeyy** [comment]
Continued here: https://github.com/ls1intum/Artemis/pull/10071

**Pull request closed without merging**